File: scratch/check_keys.js

const fs = require('fs');
const content = fs.readFileSync('assets/js/i18n.js', 'utf8');

// This is a bit hacky but should work for this specific file structure
const languages = ['en', 'hi', 'gu', 'mr'];
const keysByLang = {};

languages.forEach(lang => {
  const regex = new RegExp(`${lang}: \\{([\\s\\S]*?)\\},`, 'g');
  const match = regex.exec(content);
  if (match) {
    const block = match[1];
    const keys = block.match(/^\s*(\w+):/gm).map(k => k.trim().replace(':', ''));
    keysByLang[lang] = keys;
  }
});

const enKeys = keysByLang['en'];
languages.forEach(lang => {
  if (lang === 'en') return;
  const currentKeys = keysByLang[lang];
  const missing = enKeys.filter(k => !currentKeys.includes(k));
  if (missing.length > 0) {
    console.log(`Missing keys in ${lang}:`, missing);
  }
});

File: scratch/check_keys.py

import re

with open('assets/js/i18n.js', 'r') as f:
    content = f.read()

languages = ['en', 'hi', 'gu', 'mr']
keys_by_lang = {}

for lang in languages:
    # Match the block for each language
    pattern = rf'{lang}: \{{(.*?)\}},'
    match = re.search(pattern, content, re.DOTALL)
    if match:
        block = match.group(1)
        # Match keys at the start of lines
        keys = re.findall(r'^\s*(\w+):', block, re.MULTILINE)
        keys_by_lang[lang] = set(keys)

en_keys = keys_by_lang.get('en', set())
for lang in languages:
    if lang == 'en': continue
    current_keys = keys_by_lang.get(lang, set())
    missing = en_keys - current_keys
    if missing:
        print(f"Missing keys in {lang}: {sorted(list(missing))}")
    
    extra = current_keys - en_keys
    if extra:
        print(f"Extra keys in {lang}: {sorted(list(extra))}")
